core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -173,6 +179,16 @@ public synchronized void connect() throws IOException, ConnectionFailedException
         afterConnect();
         checkVersion(getClient(), getServiceVersion());
         return;
+      } catch (IOException e) {
+        if (FRAME_SIZE_EXCEPTION_PATTERN.matcher(e.getMessage()).find()) {
+          // See an error like "Frame size (67108864) larger than max length (16777216)!",
+          // pointing to the helper page.
+          LOG.error("Failed to set up a connection to " + getServiceName() + " "
```

Comment:
FWIW, it I also think that the "mismatched transport ..." is not very helpful without actually going to the debugging guide.

I would suggest we simple say something along the lines of:

```
The following exception was encountered when trying to establish a network connection: <exception message here>

Please consult <webpage here> to understand how to troubleshoot this error.
```

I would then use this pattern anywhere, where we want people to look at the debugging guide.